src/shogun/neuralnets/NeuralNetwork.cpp

```diff
@@ -354,6 +366,27 @@ bool CNeuralNetwork::train_gradient_descent(SGMatrix<float64_t> inputs,
 	return true;
 }
 
+void CNeuralNetwork::initialize_descend_updater()
```

Comment:
Could we pass an object instead? It is much cleaner and we won't have any need to maintain the enums